File: connection-backend/src/main/java/de/florianschmitt/service/BasePageableAdminService.java
package de.florianschmitt.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.io.Serializable;

public interface BasePageableAdminService<ENTITY extends Serializable> extends BaseAdminService<ENTITY> {
    Page<ENTITY> findAll(Pageable pageable);
}

File: src/main/kotlin/de/florianschmitt/service/BasePageableAdminService.kt
package de.florianschmitt.service

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

import java.io.Serializable

interface BasePageableAdminService<ENTITY : Serializable> : BaseAdminService<ENTITY> {
    fun findAll(pageable: Pageable): Page<ENTITY>
}
